File: src/main/java/com/github/zyypj/tadeuBooter/minecraft/tool/location/geographic/SimpleLocation.java
package com.github.zyypj.tadeuBooter.minecraft.tool.location.geographic;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.bukkit.World;

@Data
@AllArgsConstructor
public class SimpleLocation {
    private final String worldName;
    private final int x;
    private final int y;
    private final int z;

    /**
     * Construtor que recebe um objeto {@link World}.
     *
     * @param world Objeto World.
     * @param x   Coordenada X.
     * @param y   Coordenada Y.
     * @param z   Coordenada Z.
     */
    public SimpleLocation(World world, int x, int y, int z) {
        this(world.getName(), x, y, z);
    }
}

File: spigot/src/main/java/me/zyypj/booter/minecraft/spigot/location/geographic/SimpleLocation.kt
package me.zyypj.booter.minecraft.spigot.location.geographic

import org.bukkit.World

/**
 * Representa uma localização simples com nome do mundo e coordenadas inteiras.
 * Suporta serialização para string e desserialização.
 */
data class SimpleLocation(
    val worldName: String,
    val x: Int,
    val y: Int,
    val z: Int
) {
    /**
     * Construtor alternativo para instanciar diretamente a partir de um objeto World.
     */
    constructor(world: World, x: Int, y: Int, z: Int) : this(world.name, x, y, z)

    /**
     * Converte esta instância em string no formato: worldName x y z
     */
    fun serialize(): String = listOf(worldName, x, y, z).joinToString(" ")

    companion object {
        private const val SPACE_PLACEHOLDER = "-"

        /**
         * Fábrica estática para chamadas Java a partir de World e coordenadas.
         */
        @JvmStatic
        fun from(world: World, x: Int, y: Int, z: Int): SimpleLocation =
            SimpleLocation(world.name, x, y, z)

        /**
         * Constrói uma instância a partir de string serializada no formato: worldName x y z
         */
        @JvmStatic
        fun deserialize(data: String): SimpleLocation {
            val parts = data.split(" ")
            return SimpleLocation(
                parts[0].replace(SPACE_PLACEHOLDER, " "),
                parts[1].toInt(),
                parts[2].toInt(),
                parts[3].toInt()
            )
        }
    }
}